pkg/logql/metrics.go

```diff
@@ -243,6 +244,37 @@ func PrintMatches(matches []string) string {
 	return strings.Join(matches, ":")
 }
 
+func RecordIngesterSeriesQueryMetrics(ctx context.Context, log log.Logger, start, end time.Time, query string, status string, stats logql_stats.Result) {
+	var (
+		logger      = fixLogger(ctx, log)
+		latencyType = latencyTypeFast
+		queryType   = QueryTypeIngesterSeries
+	)
+
+	// Tag throughput metric by latency type based on a threshold.
+	// Latency below the threshold is fast, above is slow.
+	if stats.Summary.ExecTime > slowQueryThresholdSecond {
+		latencyType = latencyTypeSlow
+	}
+
+	logValues := make([]interface{}, 0, 11)
+	logValues = append(logValues,
+		"latency", latencyType,
```

Comment:
Would be useful to log the number of lines here, too
If we have the bytes returned as well that'd be very valuable

Reply:
Good idea as well. I've added as many as possible.
Note that the nr of lines was stored in the stats type before. Now it's a separate field to make it consistent with the other logging functions.